modules/managers/build-locally.sh

```diff
@@ -250,8 +250,20 @@ function update_release_yaml {
     fi
 }
 
+function check_if_script_invoked_directly() {
+    # Check if the script is being called directly or from another script
+    if [[ -z "${IN_CHAIN_MODE}" ]]; then
```

Comment:
I prefer the script to have a new flag. `--detect-secrets true` or `--detect-secrets false` with the default to be true, but when the uber-build-tool calls it, it can use false ? Rather than this hidden IN_CHAIN_MODE env variable...